### ISSUE TITLE
Last position in note not remembered in Desktop apps since 2.13.15

### ISSUE THREAD

**Issue by benftrex** [laurent22/joplin#10195]
### Operating system

macOS

### Joplin version

2.14.20

### Desktop version info

Joplin 2.13.15 (prod, darwin)

Client ID: 
Sync Version: 3
Profile Version: 44
Keychain Supported: Yes

Revision: 7d2c1c0

Backup: 1.4.0

### Current behaviour

When i go from one note to another, and THEN come back to the previous note, the position (inside the note) i was in doesn't get remembered like it used to , and always brings me to the START of the note. 
This happensin ALL desktop (mac) versions i have used since 2.13.15. 
So, if i have long notes, and i just move to another to copy something and go back, i have to again scroll back all the way to the point where i was, making Joplin almost unusable for my use case. 

### Expected behaviour

When i go back to a previously opened note in that session, it shoould bring me to the last position i was in that note, and not to the top of the note. 
This is how it always was for me until 2.14.17 came (using on macs for last 2 years). 
2.13.15 is the last version where this works correctly, and for now i'm forced to use this version. I have tried the 3 stable versions that have come after that, but all have the same behaviour. 

If this is a new/wanted "feature"there should at least be an option to toggle it.

**Comment by plerohellec** [comment]
I can confirm this bug on the desktop app Linux and MacOS 2.14.20 (but yes, it's been broken for a couple of months at least). It's very very annoying on long notes where you have to make frequent edits far from the top.

**Comment by benftrex** [comment]
Yes. I usually work with large notes, and this bug makes using Joplin impossible for me. I'm still stuck with the last working version (2.13.15).

**Comment by benftrex** [comment]
Can anyone confirm if this bug on desktop apps still persists in the latest versions , or it's still the same ?
Thank you

**Comment by gmilowamw** [comment]
This is still an issue for me on macOS

**Comment by plerohellec** [comment]
Still an issue on both MacOS and Linux with 2.14.20 for me.

**Comment by plerohellec** [comment]
An additional bit of information: the bug does not happen for me in the "Viewer" layout. But it does systematically happen when in the "Editor" layout at the moment of the switch. It should be extremely easy to reproduce.

**Comment by ahxxm** [comment]
can confirm on windows, 2.14.19

Editor(markdown) mode consistently reverts to the top, "Viewer" layout works fine

**Comment by benftrex** [comment]
Tried again on Mac, with both 2.14.22 and 3.0.10 (pre release) and the bug still persists. 
As others have said, the position is remembered ONLY in "viewer"layout, but always goes the top in editor mode. 

Last version where it also works in editor mode is still 2.13.15

**Comment by sakkarose** [comment]
Happened on Windows 11
```
Joplin 3.0.13 (prod, win32)

Client ID: c1e1a0805e7f43c08a3789232b0b26ad
Sync Version: 3
Profile Version: 47
Keychain Supported: Yes

Revision: 599cf5b

Backup: 1.4.1
Conflict Resolution: 1.2.3
Draw.io: 2.2.0
Kanban: 1.1.0
Menu items, Shortcuts, Toolbar icons: 1.1.0
Note list and sidebar toggle buttons: 1.0.3
Quick Links: 1.3.2
Templates: 2.4.0
```

**Comment by dinayuil** [comment]
the problem still happened on my windows 10

> Joplin 3.0.14 (prod, win32)
> 
> 客户端 ID: 2c5e20bd14254401baa0e6c3beb6f825
> 同步版本：3
> 配置文件版本：47
> 支持的密钥链：是
> 
> 修订：05cf51e
> 
> Backup: 1.4.1
> Note Tabs: 1.4.0
> Outline: 1.5.13
> Rich Markdown: 0.15.0
> Search & Replace: 2.2.0
> Space Indenter: 0.2.5
> Table Formatter Plugin: 1.2.1
> Whitespacer: 1.0.1

**Comment by benftrex** [comment]
Can anyone from the Joplin team at least confirm that they are aware of this issue/bug ?  Or if this will be a wanted "feature" from now on ? Thanks

**Comment by personalizedrefrigerator** [comment]
**Editor-only mode**: I can reproduce this if only the Markdown editor is visible (Joplin v3.1.1).

**Rich Text Editor and Markdown with viewer**: It seems to be partially working for me in the Rich Text editor and Markdown editor in Joplin v3.1.1. If I do the following, scroll position seems to be saved:
1. Create two long notes (with scroll).
2. Scroll to the end of the first note and type.
3. Switch to the second note.
4. Switch back to the first note.

However, if I then type, 
- in the Rich Text Editor, the cursor jumps to the top of the note, and,
- in the Markdown editor, the cursor jumps to somewhere in the middle of the note (the position the cursor was in the previous note).


<blockquote>
Joplin 3.1.1 (prod, linux)

Client ID: 511ce522c2a74e568d33524668d23fdc
Sync Version: 3
Profile Version: 47
Keychain Supported: Yes

Revision: a97c04c

Backup: 1.4.2
</blockquote>

**Comment by milowg** [comment]
I found an alternative solution if anyone is interested. You can use [Obsidian](https://obsidian.md/) with the `Remember cursor position` Community Plugin. Obsidian works similar to Joplin and even has some extra features like tabs. With the plugin, it remembers my scroll position on every note.

**Comment by ahxxm** [comment]
> an alternative solution

very droll
- - -
Assuming that 
- Joplin codes introduced the bug instead of dependency version bumps
- v2.14.1 is the earliest version that breaks

Then bad commit should appear in [commits from v2.13.15 to v2.14.1](https://github.com/laurent22/joplin/compare/v2.13.15...v2.14.1), does any commit looks suspicious to you?

**Comment by executed** [comment]
@ahxxm good info! 
If only someone with a local development setup could git checkout v2.14.1, and make git reset HEAD by one commit at a time until v2.13.15 that would be super useful.
Knowing the exact commit that caused the issue would be a great place to start.

**Comment by executed** [comment]
On Joplin v3.3.4 (pre-release) I can confirm personalizedrefridgerator observations.
Specifically markdown editor+viewer setup - when going back to previous note both viewer and editor are scrolled to valid historical position, but when starting typing smth -> cursor ends up in different location.

**Comment by benftrex** [comment]
Just tried with v3.3.6, and at least for me, it goes back again to NOT remembering any previous position when changing notes. Not even in the manner explained above by @executed . It just always goes to the top of each note. 

I'm still using the more than yr old v2.13.15 on my macs. 


> [@ahxxm](https://github.com/ahxxm) good info! If only someone with a local development setup could git checkout v2.14.1, and make git reset HEAD by one commit at a time until v2.13.15 that would be super useful. Knowing the exact commit that caused the issue would be a great place to start.

It would be great if someone could do something like that, and finally try yo fix this bug.

**Comment by executed** [comment]
I recommend to try out "[Resume Note](https://github.com/alondmnt/joplin-plugin-resume-note)" plugin as workaround - it aims to fix this bug - works really good in most of the cases.

In plugin settings increase "Delay before setting cursor and scroll position (in ms)" to 1000.

**Comment by benftrex** [comment]
> I recommend to try out "[Resume Note](https://github.com/alondmnt/joplin-plugin-resume-note)" plugin as workaround - it aims to fix this bug - works really good in most of the cases.

Thanks, but it's not working for me in my test with v3.3.6.
 Installed the plugin, but the bug still persists for me : 
I go to note A, and write something at the bottom of the note. Then i go to note B, and to the same. Then when i go back to note A, it doesn't bring me back to the bottom of the note (where i last wrote) like it was the case until v2.13.15. It just brings me to a random position in the middle of the note. 
The only difference the plugin does for me: it doesn't go to the top of the note, just some random position.

**Comment by executed** [comment]
@benftrex in plugin settings increase "Delay before setting cursor and scroll position (in ms)" to 1000.
We're going to make 1000 a default value in next release.

Also cross-note link navigation to specific note section is broken now - track this [PR](https://github.com/alondmnt/joplin-plugin-resume-note/pull/4).

**Comment by benftrex** [comment]
Hi, 
It's been now more than a year since this bug first appeared. It still hasn't been fixed. Until now i could still use Joplin , by using older versions on my Macs (2.13.15, last working), and the newer ones on Android and Ipad (since i don't need that function so much there). 
But since the last Android app update i have a new problem: If i edit a note on Android's newest version (3.4.5), i then CAN'T see that same note on my macs. It says there's some kind of error with the encryption,  and the note can't be decrypted. 
I suppose something has changed in newer versions of the mobile app, that change how the note gets encrypted/decrypted.

This makes Joplin definitely unusable for me. I really need the position to be remembered in my desktop apps, since i hop alot from note to note, and copy/paste from different notes. 

the "resume Note" plugin doesn't work at all for me. So i've reached now the point where i sadly need an alternative to Joplin. 
I know this is not the place to ask, but i'd really appreciate any suggestion. I currently self-host a joplin server with docker, and use it on Macs and Android mostly. I tried Standard notes in the past, but would like to know if other alternatives or newer exist? 

Thank you

**Comment by executed** [comment]
@benftrex 
1. This is not related "If i edit a note on Android's newest version (3.4.5), i then CAN'T see that same note on my macs.";
2. "resume Note" plugin should be working, if it's not - create new issue (https://github.com/alondmnt/joplin-plugin-resume-note).
